perfdash/parser.go

```diff
@@ -228,3 +228,46 @@ func pareseSchedulingThroughput(data []byte, buildNumber int, testResult *BuildD
 	perfData.Data["Average"] = obj.ThroughputAverage
 	testResult.Builds[build] = append(testResult.Builds[build], perfData)
 }
+
+// TODO(krzysied): This structure also should be moved to metric package.
+type Histogram struct {
+	Labels  map[string]string `json:"labels"`
+	Buckets map[string]int    `json:"buckets"`
+}
+
+type HistogramVec []Histogram
+
+// This structure differs from the one from metric_util.go
+// Instead of explicit fields, we have a map.
+type EtcdMetrics map[string]HistogramVec
+
+func parseEtcdMetric(metricName string) func(data []byte, buildNumber int, testResult *BuildData) {
```

Comment:
Can you change this function so that we will be able to use this in general for all histograms (not just etcd)?